src/compose.ts

```diff
@@ -346,3 +381,175 @@ function normalizeKeyValuePairs(
 		.fromPairs()
 		.value();
 }
+
+/**
+ * Reads a normalized composition object and expands context of composition
+ * Context is related to the composition in terms of env_file tag in composition
+ * env_file tag values (file paths) are normalized and validated.
+ *
+ * Caches environment files for all services in the composition and transfers all env_file
+ * environment vars into the environment tag of the composition.
+ *
+ * Drops env_file propertie from composition to indicate that the expand has taken place
+ *
+ * @param normalizedComposition normalized composition object
+ * @param fileResolverCb callback from caller that returns a readStream for a given filePath string
+ * @returns Composition with expanded context
+ */
+async function expandContext(
+	normalizedComposition: Composition,
+	fileResolverCb: (path: string) => Promise<ReadStream>,
+): Promise<Composition> {
+	// read all env_file delcared file paths from the composition
+	// normalize the environment variables
+	const expandedEnvrionmentFiles: Dict<Dict<string>> =
```

Comment:
`expandedEnvrionmentFiles` spelling mistake in `Envrionment`

Reply:
Thanks, corrected